app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ReservationsController < ApplicationController
+
+  def create
+    @property = Property.find_by(id: params[:property_id])
+
+    start_date = Date.strptime(params[:reservation]['start_date'], "%m/%d/%Y")
+    end_date = Date.strptime(params[:reservation]['end_date'], "%m/%d/%Y")
+
+    @reservation = current_user.reservations.new(start_date: start_date, end_date: end_date)
+    @reservation.property = @property
+
+    if current_user == @property.user
+      redirect_to @property, error: 'Cannot book own property!'
+    elsif @reservation.save
+      redirect_to @property, success: 'Booked successfully'
+    else
+      redirect_to @property, error: 'Failed to book'
+    end
+  end
+
+  def index
+    @properties = current_user.properties
+    @count = 0
```

Comment:
Why is this `count` needed?

Reply:
The count is just there for the sake of determining when to show a message of "The page you are looking for is empty" message if the user made a reservation or received a reservation. So, if the reservation count is 0, then it will show the message,  "The page you are looking for is empty".